tests/functional/behave_features/common/utils/workflow_repo_manager.py

```diff
@@ -0,0 +1,168 @@
+
+        Args:
+            branch_name: The branch name to create.
+
+        Raises:
+           Exception: An error occurred creating the branch.
+        """
+        try:
+            self.repo.git.checkout("-b", branch_name)
+        except Exception as e:
+            raise Exception(f"Unable to create branch: {e}")
+
+        self.__local_branches_created.append(branch_name)
+
+    def add_worktree(self) -> TemporaryDirectory:
+        """Creates a worktree at a generated tempdir.
+
+        Worktrees created with this method are stored internally
+        and cleaned up when cleanup() is called.
+
+        Returns:
+            The TemporaryDirectory for the worktree that was requested.
+
+        Raise:
+            Exception: An error occurred when creating the worktree.
+        """
+        worktree_dir = TemporaryDirectory(prefix="worktree-")
+
+        try:
+            self.repo.git.worktree("add", "--detach", worktree_dir.name, "HEAD")
+        except Exception as e:
+            raise Exception(f"Unable to create worktree: {e}")
+
+        self.__local_worktrees_created.append(worktree_dir)
+
+        return worktree_dir
+
+    def cleanup_local_branches(self):
+        """Cleans up branches created by this repo manager."""
+        for br in self.__local_branches_created:
+            try:
+                logging.info(f'Cleaning up generated local branch: "{br}"')
+                self.repo.git.branch("-D", br)
+            except git.exc.GitCommandError:
+                logging.warn(
+                    f'local branch "{br}" could not be deleted, potentially because it did not exist'
+                )
+        self.__local_branches_created = []
+
+    def cleanup_worktrees(self):
+        """Cleans up local worktrees created by this repo manager."""
+        for wt in self.__local_worktrees_created:
+            logging.info(f'Cleaning up generated local worktree: "{wt}"')
+            try:
+                self.repo.git.worktree("remove", wt.name)
+            except git.exc.GitCommandError:
+                logging.warn(
+                    f'local worktree "{wt}" could not be deleted, potentially because it did not exist'
+                )
+        self.__local_worktrees_created = []
+
+    def cleanup_remote_branches(self):
+        """Cleans up remote branches created by this repo manager."""
+        for rbr in self.__remote_branches_created:
+            remote = rbr[0]
+            branch = rbr[1]
+            logging.info(f'Cleaning up branch "{branch}" from remote "{remote}')
+            try:
+                github.github_api(
+                    "delete",
+                    f"repos/{remote}/git/refs/heads/{branch}",
+                    self.__authtoken,
+                )
+            except git.exc.GitCommandError:
+                logging.warn(
+                    f'remote branch "{branch}" could not be deleted, potentially because it did not exist'
+                )
```

Comment:
Yep, good catch. I generally preferred using the API for things like this, but only because I felt like it was a safer bet that the command would use _stored credentials_ and not _local privileges_ (e.g. SSH keys authorized for git pushes).

With that said, `git` seems perfectly happy accepting the remote reference in this format, so I'll go with it. This has been updated, along with the accompanying exception(s).

Reply:
Could you double check if you changed this in the intended way ? From your command, I understood you wanted to use the `git` module ?

Reply:
Ah sorry, this was modified in the next commit again, all good